py3status/modules/window_manipulation.py

```diff
@@ -0,0 +1,396 @@
+window_manipulation {
+    button_focus = 1
+    button_next = 5
+    button_previous = 4
+
+    format = '{format_window} '
+    format += '[\?color=#f00&show DIVIDER] '
+    format += '[\?color=#0ff&show NAME:] {name} '
+    format += '[\?color=#0ff&show CLASS:] {class} '
+    format += '[\?color=#0ff&show FLOATING:] {floating} '
+    format += '[\?color=#0ff&show SCRATCHPAD:] {scratchpad} '
+    format += '[\?color=#0ff&show URGENT:] {urgent} '
+    format += '[\?color=#0ff&show URGENT_SP:] {urgent_sp} '
+    format += '[\?color=#0ff&show WINDOW:] {window}'
+    format += '[\?color=#0ff&show INDEX:] {index} '
+    format += '[\?color=#0ff&show ID:] {id} '
+
+    format_window = '\?max_length=30 [\?if=is_urgent&color=urgent \u2756'
+    format_window += '|[\?if=is_focused&color=focused \u2756'
+    format_window += '|[\?if=is_scratchpad&color=scratchpad \u2756'
+    format_window += '|[\?if=is_floating&color=floating \u2756'
+    format_window += '|\?color=unfocused \u2756]]]] {name}'
+
+    color_floating = '#3DCCB0'
+    color_focused = '#FFAE3D'
+    color_scratchpad = '#666666'
+    color_unfocused = '#3DAEE9'
+    color_urgent = '#FF48C4'
+}
+# add your magic snippets here!
+```
+
+SAMPLE OUTPUT
+[
+    {'color': u'#FF0000', 'full_text': u'\u2022'},
+    {'color': u'#FF0000', 'full_text': u'\u2022'},
+    {'color': u'#00FF00', 'full_text': u'\u2022'},
+    {'color': u'#FF0000', 'full_text': u'\u2022'},
+    {'color': u'#FF0000', 'full_text': u'\u2022'},
+    {'color': u'#FF0000', 'full_text': u'\u2022'},
+    {'color': u'#FF0000', 'full_text': u'\u2022'},
+    {'color': u'#FF0000', 'full_text': u'\u2022'},
+    {'color': u'#FF0000', 'full_text': u'\u2022'},
+]
+
+old_default
+[
+    {'full_text': u'\u2756 Netflix '},
+    {'full_text': u'\u2756 #py3status - irssi'},
+    {'full_text': u'\u2756 z420: ~ '},
+    {'full_text': u'\u2756 ultrabug/py3status: py3statu '},
+    {'full_text': u'\u2756 window_manipulation.py (~/src/py3sta '},
+    {'full_text': u'\u2756 mpd_status.py (~/src/py3stat '},
+    {'full_text': u'\u2756 (10) Inbox - Firefox'},
+]
+
+colors
+[
+    {'color': u'#3DAEE9', 'full_text': u'\u2756'},
+    {'full_text': u' Netflix '},
+    {'color': u'#FF48C4', 'full_text': u'\u2756'},
+    {'full_text': u' #py3status - irssi'},
+    {'color': u'#3DAEE9', 'full_text': u'\u2756'},
+    {'full_text': u' z420: ~ '},
+    {'color': u'#3DAEE9', 'full_text': u'\u2756'},
+    {'full_text': u' ultrabug/py3status: py3statu '},
+    {'color': u'#FFAE3D', 'full_text': u'\u2756'},
+    {'full_text': u' window_manipulation.py (~/src/py3sta '},
+    {'color': u'#3DAEE9', 'full_text': u'\u2756'},
+    {'full_text': u' mpd_status.py (~/src/py3stat '},
+    {'color': u'#3DAEE9', 'full_text': u'\u2756'},
+    {'full_text': u' (10) Inbox - Firefox'},
+]
+
+"""
+
+from threading import Thread
+
+import i3ipc
+
+
+class Py3status:
+    """
+    """
+    # available configuration parameters
+    allow_urgent = None
```

Comment:
> Also the description was not clear to the functionality

Take a guess. :-) Okay, I'll update the description.

* `window` allows all urgents including ordinary `scratchpad` workspace.
* `scratchpad` allows urgent in scratchpad only. This is so we can behave like scratchpad properly without getting an urgent for something that happens in `window` (eg any workspace).

Maybe... `all`, `scratchpad` would be better.

What if I start off checking for a boolean instance... Otherwise, strings... Would that work okay? It's off by default (ie None) because it's ugly and unproductive to support `urgent` by default... is which why I included a fake `urgent` example.